packages/pwa-kit-cli/src/ssr/server/build-dev-server.js

```diff
@@ -173,6 +146,30 @@ export const DevServerMixin = {
         })
     },
 
+    serveServiceWorker(req, res) {
+        req.app.__devMiddleware.waitUntilValid(() => {
+            const sourceMap = req.path.endsWith('.map')
+            const file = sourceMap ? 'worker.js.map' : 'worker.js'
+            const type = sourceMap ? '.js.map' : '.js'
+            const content = DevServerFactory._getWebpackAsset(req, 'client-optional', file)
+            if (content === null) {
+                res.sendStatus(404)
+            } else {
+                res.type(type)
+                res.send(content)
+            }
+        })
+    },
+
+    render(req, res, next) {
+        const app = req.app
+        if (app.__webpackReady()) {
+            app.__hotServerMiddleware(req, res, next)
```

Comment:
@vmarta This is the place that might conflict with your pr's change.

Reply:
Thanks @bendvc for the heads up. My change is now simpler, so the merge conflict would be easier to resolve.